src/core/ext/filters/client_channel/lb_policy/round_robin/round_robin.c

```diff
@@ -74,6 +74,9 @@ typedef struct round_robin_lb_policy {
   bool started_picking;
   /** are we shutting down? */
   bool shutdown;
+  /** has the policy gotten into the GRPC_CHANNEL_SHUTDOWN? No picks can be
+   * service after this point, the policy will never transition out. */
+  bool in_connectivity_shutdown;
```

Comment:
Done. Tests are happy. I'm of two minds about this. On the one hand, it avoid having another bool. On the other, it conflates two concerns on a single bool: the actual shutdown that happens when all strong refs are gone and the "pseudo" shutdown induced by the transition into SHUTDOWN wrt connectivity.